src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -2213,6 +2216,36 @@ def set_frame_2d_labels(
     index_is_frame = isinstance(index, InternalFrame)
     item_is_frame = isinstance(item, InternalFrame)
     item_is_scalar = is_scalar(item)
+    original_index = index
+    # If `item` is from a Series (rather than a Dataframe), flip the series item values to apply them
+    # across columns rather than rows.
+    if frame_is_df_and_item_is_series and (columns == slice(None) or len(columns) > 1):  # type: ignore[arg-type]
+        # If columns is slice(None), we are setting all columns in the InternalFrame.
+        matching_item_columns_by_label = True
+        col_len = (
+            len(internal_frame.data_column_snowflake_quoted_identifiers)
+            if columns == slice(None)
+            else len(columns)  # type: ignore[arg-type]
+        )
+        item = get_item_series_as_single_row_frame(
+            item, col_len, move_index_to_cols=True
+        )
+
+        if is_scalar(index):
```

Comment:
what happens if index is not scalar?

Reply:
If index is not scalar, we don't have to append it to the item to match index - it should either be slice(None) or an internalframe, which we handle in the rest of the method.